dask/array/linalg.py

```diff
@@ -573,8 +577,15 @@ def solve(a, b):
         Solution to the system ``a x = b``.  Shape of the return matches the
         shape of `b`.
     """
-    p, l, u = lu(a)
-    uy = solve_triangular(l, p.T.dot(b), lower=True)
+    if sym_pos:
+        u = cholesky(a)
+        lname = u.name.replace('cholesky-upper-', 'cholesky-')
```

Comment:
Mutating the name here worries me slightly. Is there any way to get around doing this? Perhaps move the logic in `cholesky` to `_cholesky`, which allows you to create the graph you want here directly, and then in the user facing `cholesky` don't expose that option?